feat(datasource/github-runners): add macos 14 beta (#27292)
renovatebot/renovate: lib/modules/datasource/github-runners/readme.md

This datasource returns a list of all runners that are hosted by GitHub.
The datasource is based on [GitHub's `runner-images` repository](https://github.com/actions/runner-images).

> Examples: `windows-2019` / `ubuntu-22.04` / `macos-13`

Reviewer: ```suggestion
Examples: `windows-2022` / `ubuntu-22.04` / `macos-13`
```

2022 is the latest stable version for Windows, according to the upstream readme. So I'm updating this reference in our docs.

Ubuntu 24.04 will become latest soon, but it'll update our docs myself then. 😄

## Maintenance

New runner versions must be added to the datasource with a pull request.
Unstable runners are tagged as `[beta]` in the readme of the [`runner-images` repository](https://github.com/actions/runner-images) and should have the `isStable:false` property.

> Deprecated runners are tagged as `[deprecated]` in the readme and should have the `isDeprecated:true` property.

Reviewer: Looks like the upstream readme drops really old runner versions. Because they used to have a Ubuntu 16.04 runner, but the readme does not mention it (anymore).

I think we should have a line to capture that problem:

```suggestion
Deprecated runners are tagged as `[deprecated]` in the readme and should have the `isDeprecated:true` property.
If a runner is very old, the readme may drop it completely, but we should still give it the `isDeprecated:true` property.
```

Once a runner version becomes stable, the `[beta]` tag is removed and the suffix `latest` is added to its YAML label and the `isStable:false` should be removed.

> Once a runner version becomes deprecated, the `isDeprecated:true` property should be added.

Reviewer: After you apply my suggestion above, can you add new `h3` headings, and move the text to the correct place? Here's what I mean:

```
## Maintenance

### Adding a new beta version

### Marking a version as stable

### Deprecating a version
```

Author: can you please make another suggestion now i've merged yours 🤗

This datasource returns a list of all runners that are hosted by GitHub.
The datasource is based on [GitHub's `runner-images` repository](https://github.com/actions/runner-images).

Examples: `windows-2022` / `ubuntu-22.04` / `macos-13`

## Maintenance

> New runner versions must be added to the datasource with a pull request.
> Unstable runners are tagged as `[beta]` in the readme of the [`runner-images` repository](https://github.com/actions/runner-images) and should have the `isStable:false` property.
> Deprecated runners are tagged as `[deprecated]` in the readme and should have the `isDeprecated:true` property.
> If a runner is very old, the readme may drop it completely, but we should still give it the `isDeprecated:true` property.
> Once a runner version becomes stable, the `[beta]` tag is removed and the suffix `latest` is added to its YAML label and the `isStable:false` should be removed.
> Once a runner version becomes deprecated, the `isDeprecated:true` property should be added.

Reviewer: ```suggestion
### Adding a new version

New runner versions must be added to the datasource with a pull request.

#### Unstable runners

Unstable runners are tagged as `[beta]` in [the `runner-images` repository's readme](https://github.com/actions/runner-images) and should get the `isStable:false` property in our code.

#### Promoting a version to stable

Once a runner version becomes stable, the `[beta]` tag is removed and the suffix `latest` is added to its YAML label.
We then remove the `isStable:false` property in our code.

### Deprecating a version

Deprecated runners are tagged as `[deprecated]` in [the `runner-images` repository's readme](https://github.com/actions/runner-images) and we should give it the `isDeprecated:true` property.
If a runner is very old, the readme may drop it completely, but we should still give it the `isDeprecated:true` property.
```
